testnet/stacks-node/src/burnchains/bitcoin_regtest_controller.rs

```diff
@@ -1918,22 +1928,21 @@ impl BitcoinRegtestController {
 
     /// Send a serialized tx to the Bitcoin node.  Return Some(txid) on successful send; None on
     /// failure.
-    pub fn send_transaction(&self, transaction: SerializedTx) -> Option<Txid> {
-        debug!("Send raw transaction: {}", transaction.to_hex());
-        let result = BitcoinRPCRequest::send_raw_transaction(&self.config, transaction.to_hex());
-        match result {
-            Ok(_) => {
-                debug!("Sent transaction {}", &transaction.txid);
-                Some(transaction.txid())
-            }
-            Err(e) => {
-                error!(
-                    "Bitcoin RPC failure: transaction submission failed - {:?}",
-                    e
-                );
-                None
-            }
-        }
+    pub fn send_transaction(
+        &self,
+        transaction: SerializedTx,
+    ) -> Result<Txid, BurnchainControllerError> {
+        debug!("Sending raw transaction: {}", transaction.to_hex());
+
+        BitcoinRPCRequest::send_raw_transaction(&self.config, transaction.to_hex())
+            .map(|_| {
+                debug!("Transaction {} sent successfully", &transaction.txid());
+                transaction.txid()
+            })
+            .map_err(|e| {
+                error!("Bitcoin RPC error: transaction submission failed - {:?}", e);
+                BurnchainControllerError::TransactionSubmissionFailed
```

Comment:
Can `TransactionSubmissionFailed` capture the inner error here?

Reply:
Are you okay with it just being a `String`, since `RPCError` is inside this implementation of the trait?

Reply:
yeah that's fine

Reply:
Addressed in ca885d07e101fd7bbe8db0c4b6ca62cc23dc1033